documentation/reference/ci_cd.md

```diff
@@ -0,0 +1,491 @@
+This job is treated as the proof of functionality for publishing Docker images
+for the ingestion server.
+```
+
+## API jobs
+
+### `test-api`
+
+Initialises the API using the `api/init` recipe and runs tests for the API using
+the `api/test` recipe. Tests are run inside a Docker container so neither Python
+nor Node.js needs to be installed.
+
+This job is skipped if the API codebase has not changed. Its counterparts are
+[`test-cat`](#test-cat) for the catalog, [`test-ing`](#test-ing) for the
+ingestion server and [`nuxt-build`](#nuxt-build) for the frontend.
+
+```{note}
+This job is treated as the proof of functionality for publishing Docker images
+for the API. Since it also initialises the API, it also largely proves that the
+ingestion server is working.
+```
+
+### `django-checks`
+
+Runs a matrix of various Django checks for the API using the following `just`
+recipes.
+
+- `api/dj check`
+- `api/dj validateopenapischema`
+- `api/dj makemigrations --check --noinput --merge`
+- `api/doc-test`
+
+This job is skipped if the API codebase has not changed. Its counterpart is
+[`nuxt-checks`](#nuxt-checks) for the frontend.
+
+#### `bypass-django-checks`
+
+The `django-checks` job is marked as required check in GitHub. If the job is
+skipped, the matrix will not be expanded into individual jobs and all required
+checks will appear as pending, thus blocking PRs from being merged.
+
+To counter this, the `bypass-django-checks` job is run if the API codebase has
+not changed, and it simply passes the checks by having the same job names.
+
+## Frontend jobs
+
+### `nuxt-build`
+
+```{note}
+This job is treated as the proof of functionality for publishing Docker images
+for the frotend.
+
+This job, combined with the [`playwright` job](#playwright), is treated as the
+proof of functionality for deploying frontend to staging.
+```
+
+### `nuxt-checks`
+
+Runs a matrix for various Nuxt checks for the frontend using the following
+Node.js scripts.
+
+- `test:unit`
+- `storybook:smoke`
+
+This job is skipped if the frontend codebase has not changed. Its counterpart is
+[`django-checks`](#django-checks) for the API.
+
+#### `bypass-nuxt-checks`
+
+The `nuxt-checks` job is marked as required check in GitHub. If the job is
+skipped, the matrix will not be expanded into individual jobs and all required
+checks will appear as pending, thus blocking PRs from being merged.
+
+To counter this, the `bypass-nuxt-checks` job is run if the frontend codebase
+has not changed, and it simply passes the checks by having the same job names.
+
+### `playwright`
+
+Runs a matrix of various Playwright tests for the frontend using the following
+Node.js scripts.
+
+- `test:playwright visual-regression`
+- `test:playwright e2e`
+- `test:storybook`
+
+This job is skipped if the frontend codebase has not changed.
+
+```{note}
+This job, combined with the [`nuxt-build` job](#nuxt-build), is treated as the
+proof of functionality for deploying frontend to staging.
+```
```

Comment:
Why are the `nuxt-checks` not required for deployment?

Reply:
Unit tests were deemed not that essential for deployment because
- the `lint` job (which is required for basically everything) includes type checking with `vue-tsc`
- the `nuxt-build` job (which is required for deploy) proves that Nuxt doesn't have any major issues to build and run
- the `playwright` job (which is required for deploy) includes the e2e tests

Reply:
Gotcha! That is definitely worth documenting in this document.

I am sceptical that it is the right choice though, but that's a different issue altogether.